tls/s2n_cipher_suites.c

```diff
@@ -921,12 +921,23 @@ static int s2n_cipher_is_compatible_with_cert(struct s2n_cipher_suite *cipher, s
     return 0;
 }
 
-static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t * wire, uint32_t count, uint32_t cipher_suite_len)
+static struct s2n_cert_chain_and_key *s2n_get_compatible_cert_chain_and_key(struct s2n_connection *conn, struct s2n_cipher_suite *cipher_suite)
 {
-    /* Only one cert chain for now */
-    notnull_check(conn->config->cert_and_key_pairs);
-    struct s2n_cert *leaf_cert = conn->config->cert_and_key_pairs->cert_chain->head;
+    for (int i = 0; i < conn->config->num_certificates; i++) {
+        struct s2n_cert_chain_and_key *cert_chain_and_key = conn->config->cert_and_key_pairs[i];
+        struct s2n_cert *leaf_cert = cert_chain_and_key->cert_chain->head;
+        uint8_t cert_compatibility = 0;
+        GUARD_PTR(s2n_cipher_is_compatible_with_cert(cipher_suite, leaf_cert, &cert_compatibility));
+        if (cert_compatibility) {
+            return cert_chain_and_key;
+        }
+    }
+
+    return NULL;
+}
 
+static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t * wire, uint32_t count, uint32_t cipher_suite_len)
```

Comment:
nitpick: this function is setting the cipher and the cert. Is there a better name that could indicate we're doing both? It makes the code in s2n_process_client_hello a little more readable

Reply:
I'm tempted to name it something like `s2n_decide_handshake_params()` but we don't quite decide all of the handshake parameters here yet.

Reply:
will go with s2n_set_cipher_and_cert_as_tls_server, etc.

Reply:
added c03c44d